File: app/src/main/java/com/alless/foundationdemo/MainActivity.java
package com.alless.foundationdemo;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.app.AppCompatActivity;
import android.util.Log;
import android.view.View;
import android.widget.Button;

import com.alless.commonlib.base.BaseActivity;

/**
 * Created by chengjie on 2019/3/19
 * Description:
 */
public class MainActivity extends BaseActivity {

    private Button mBtn_1;


    @Override
    protected int getLayoutId() {
        return R.layout.activity_main;
    }

    @Override
    protected void initView() {
        mBtn_1 = findViewById(R.id.btn_1);
    }

    @Override
    protected void initData() {

    }

    @Override
    protected void initListener() {
        mBtn_1.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                startActivity(new Intent(MainActivity.this,TestActivity1.class));
            }
        });
    }

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(TAG, "onCreate: " );
    }
}

File: kotlindemo/src/main/java/com/alless/kotlindemo/MainActivity.kt
package com.alless.kotlindemo

import android.support.v7.app.AppCompatActivity
import android.os.Bundle

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
    }

    fun main(args: Array<String>) {
        val test1 = "hello,world"   //变量
        var test2 = 555             //常量

        println(test1)
    }

    var a: Int = 11
    var b = 22      //可省略类型

    //代码块体
    fun max1(a: Int, b: Int): Int {
        return if (a > b) a else b
    }

    //表达式体
    fun max2(a: Int, b: Int): Int = if (a > b) a else b

    //表达式体（省略返回值）
    fun max3(a: Int, b: Int) = if (a > b) a else b




}
